src/playsim/p_enemy.cpp

```diff
@@ -2370,6 +2371,20 @@ DEFINE_ACTION_FUNCTION(AActor, A_Look2)
 	return 0;
 }
 
+void GoalReached(AActor* self, AActor* oldgoal)
+{
+	if (self && self->Level)
```

Comment:
`self` doesn't need to be nulled checked since the VM ensures that this won't happen